styles/layers.h

```diff
@@ -73,12 +82,35 @@ class Compose<BLACK, L1> {
   }
 };
 
-
 template<class BASE, class ... LAYERS> struct LayerSelector {};
 template<class BASE> struct LayerSelector<BASE> { typedef BASE type; };
 template<class BASE, class L1> struct LayerSelector<AlphaL<BASE, Int<0>>, L1> { typedef L1 type; };
-template<class BASE, class L1> struct LayerSelector<BASE, L1> { typedef Compose<BASE, L1> type; };
-template<class BASE, class L1, class ... REST> struct LayerSelector<BASE, L1, REST...> {
+
+// If base is exactly AlphaL<*, Int<0>> and there are 3+ args, drop it early
+template<class BASE, class L1, class L2, class... REST>
+struct LayerSelector<AlphaL<BASE, Int<0>>, L1, L2, REST...> {
+  typedef typename LayerSelector<L1, L2, REST...>::type type;
+};
+
+// Solid+solid check
+template<class BASE, class L1>
```

Comment:
Isn't it true that no layer except the first one should be solid?

Reply:
hmm. would "“Reject solid layers after the first is found" be a better comment?
This would go away anyway with a move to color.h

Reply:
Comment updated.

Reply:
I'm saying that we don't need to make so many changes, we can just insert the assert into Compose<>
Basically, if the second argument to Compose is opaque -> error

Reply:
done